public/app/features/dashboard-scene/panel-edit/getPanelFrameOptions.tsx

```diff
@@ -139,19 +140,23 @@ export function PanelDescriptionTextArea({ panel }: { panel: VizPanel }) {
 }
 
 export function PanelBackgroundSwitch({ panel }: { panel: VizPanel }) {
-  const { displayMode } = panel.useState();
+  const { displayMode = 'default' } = panel.useState();
+
+  const onChange = () => {
+    const newDisplayMode = displayMode === 'default' ? 'transparent' : 'default';
+
+    panel.publishEvent(
+      new DashboardEditActionEvent({
+        description: 'Change panel background',
+        sceneObj: panel,
+        perform: () => panel.setState({ displayMode: newDisplayMode }),
+        undo: () => panel.setState({ displayMode: displayMode }),
+      }),
+      true
+    );
+  };
```

Comment:
each state change handler will be more verbose, but think we can some up with helpers to make the above smaller for simple cases like this.

Reply:
Could we implement a helper that defaults the undo to reset the whole state of the scene object?

Reply:
@oscarkilhed maybe, it's tricky. What part of the scene state should we keep / reset? we do not want to reset nested scene objects and data state